### PR TITLE
Move to .NET 8 RTM

### PR DESCRIPTION
**Pull request by 333fred**

### CODE REVIEW

**Review by jasonmalinowski — APPROVED** [review]
image

**Comment by RikkiGibson** [comment]
I ran a signed build off of #70639. It looks like there is a publishing issue we need to work out first. https://dev.azure.com/dnceng/internal/_build/results?buildId=2316728&view=results

File: docs/contributing/Building, Debugging, and Testing on Unix.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Particularly for developers who aren't experienced with .NET Core development on
 1. Install [VS Code](https://code.visualstudio.com/Download)
     - After you install VS Code, install the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp)
     - Important tip: You can look up editor commands by name by hitting *Ctrl+Shift+P*, or by hitting *Ctrl+P* and typing a `>` character. This will help you get familiar with editor commands mentioned below. On a Mac, use *⌘* instead of *Ctrl*.
-2. Install the latest [.NET 8.0 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/8.0).
+1. Install the [.NET 8.0 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) which matches the `sdk.version` property in [global.json](../../global.json#L3)
```
Comment (jjonescz) [inline docs/contributing/Building, Debugging, and Testing on Unix.md:15]:
```suggestion
2. Install the [.NET 8.0 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) which matches the `sdk.version` property in [global.json](../../global.json#L3)
```

File: global.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
   "sdk": {
-    "version": "8.0.100-rc.1.23463.5",
+    "version": "8.0.100",
```
Comment (jjonescz) [inline global.json:3]: This should already be in main since https://github.com/dotnet/roslyn/pull/70639 automerged.

**Review by jjonescz — APPROVED** [review]

File: docs/contributing/Building, Debugging, and Testing on Unix.md (hunk 1 above)
Comment (333fred) [reply docs/contributing/Building, Debugging, and Testing on Unix.md:15]: We actually intentionally keep the number the same; markdown rendering counts up, and we don't have to worry about accidentally reordering steps.

**Pull request merged**